be/src/pipeline/pipeline_x/dependency.cpp

```diff
@@ -326,4 +330,116 @@ Status HashJoinDependency::extract_join_column(vectorized::Block& block,
     return Status::OK();
 }
 
+bool RuntimeFilterTimer::has_ready() {
+    std::unique_lock<std::mutex> lc(_lock);
+    return _runtime_filter->is_ready();
+}
+
+void RuntimeFilterTimer::call_timeout() {
+    std::unique_lock<std::mutex> lc(_lock);
+    /// TODO: check _runtime_filter->is_ready_or_timeout();
+    if (_call_ready) {
+        return;
+    }
+    _call_timeout = true;
+    if (_parent) {
+        _parent->sub_filters();
+    }
+}
+
+void RuntimeFilterTimer::call_ready() {
+    std::unique_lock<std::mutex> lc(_lock);
+    /// TODO: check _runtime_filter->is_ready_or_timeout();
```

Comment:
Do not to call `is_ready_or_timeout` because we already use dependency to replace it